omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/EncounterController1_9Test.java

```diff
@@ -500,6 +501,29 @@ public void shouldUnVoidAnEncounter() throws Exception {
 		
 	}
 	
+	@Test
+	public void shouldFetchAllPatientEncounterIncludingVoidedPatientEncountersIfIncludeAllParamIsSetToTrue()
+	        throws Exception {
+		Patient patient = Context.getPatientService().getPatientByUuid(RestTestConstants1_9.PATIENT_WITH_ENCOUNTER_UUID);
+		
+		Encounter enc = Context.getEncounterService().getEncounterByUuid(RestTestConstants1_9.ENCOUNTER_UUID);
+		Context.getEncounterService().voidEncounter(enc, "some reason");
+		Context.getEncounterService().saveEncounter(enc);
+		
+		//list of non voided encounters
+		List<Encounter> encs = Context.getEncounterService().getEncountersByPatient(patient);
+		Assert.assertEquals(1, encs.size());
+		
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("patient", RestTestConstants1_9.PATIENT_WITH_ENCOUNTER_UUID);
```

Comment:
You had more tests than this for different values of includeAll. Did you remove them accidentally?

Reply:
yes , from your comments , i perceived like you wanted me to remove those other tests. since it was the default behaviour ,before i added these changes

Reply:
here is the comment you gave regarding that other test 
`Then what is the purpose of the test? Did you read about a unit test failing first and then make changes to make it pass? TDD`

Reply:
i have re added the test again